test/attr/attr_iboutlet.swift

```diff
@@ -122,20 +110,35 @@ class NonObjC {}
   @IBOutlet var collection3b: [OP1]?
   @IBOutlet var collection3c: [OP1]!
 
-  @IBOutlet var collection4a: [CP1] // expected-error{{@IBOutlet property cannot be an array of non-'@objc' protocol type}} {{3-13=}}
   @IBOutlet var collection4b: ([CP1])? // expected-error{{@IBOutlet property cannot be an array of non-'@objc' protocol type}} {{3-13=}}
   @IBOutlet var collection4c: ([CP1])! // expected-error{{@IBOutlet property cannot be an array of non-'@objc' protocol type}} {{3-13=}}
 
   @IBOutlet var collection5b: ([String])?
   @IBOutlet var collection5c: ([String])!
 
-  @IBOutlet var collection6a: [NonObjC] // expected-error{{@IBOutlet property cannot be an array of non-'@objc' class type}} {{3-13=}}
   @IBOutlet var collection6b: ([NonObjC])? // expected-error{{@IBOutlet property cannot be an array of non-'@objc' class type}} {{3-13=}}
   @IBOutlet var collection6c: ([NonObjC])! // expected-error{{@IBOutlet property cannot be an array of non-'@objc' class type}} {{3-13=}}
 
   init() { }
 }
 
+// Check that the type is optional
+@objc class OX {
+  // expected-error@+3 {{@IBOutlet property has non-optional type 'OX'}}
+  // expected-note @+2 {{add '?' to form the optional type 'C?'}}
```

Comment:
Do you know why 'C?' is mentioned here?

Reply:
Sorry. My macOS box wasn't handy, and this file was edited without testing. I'm now actively testing on my Mac.